components/config-mgmt-service/integration_test/report_export_test.go

```diff
@@ -0,0 +1,226 @@
+package integration_test
+
+import (
+	// "bytes"
+	"context"
+	// "encoding/json"
+	"fmt"
+	"github.com/buger/jsonparser"
+	"github.com/chef/automate/api/interservice/cfgmgmt/request"
+	api "github.com/chef/automate/api/interservice/cfgmgmt/service"
+	authzConstants "github.com/chef/automate/components/authz-service/constants/v2"
+	"github.com/chef/automate/components/config-mgmt-service/backend/elastic"
+	"github.com/chef/automate/components/config-mgmt-service/config"
+	"github.com/chef/automate/components/config-mgmt-service/grpcserver"
+	iBackend "github.com/chef/automate/components/ingest-service/backend"
+	"github.com/chef/automate/lib/grpc/auth_context"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/test/bufconn"
+	"io"
+	"net"
+	"testing"
+	"time"
+)
+
+func TestReportExportProjectFilters(t *testing.T) {
+	esClient := elastic.New(elasticsearchUrl)
+	c := config.New(esClient)
+	server := grpcserver.NewCfgMgmtServer(c)
+
+	lis := bufconn.Listen(1024 * 1024)
+	s := grpc.NewServer()
+	api.RegisterCfgMgmtServer(s, server)
+
+	go func() {
+		if err := s.Serve(lis); err != nil {
+			t.Fatalf("Server exited with error: %v", err)
+		}
+	}()
+
+	dialer := func(string, time.Duration) (net.Conn, error) { return lis.Dial() }
+
+	conn, err := grpc.DialContext(context.Background(), "bufnet", grpc.WithDialer(dialer), grpc.WithInsecure())
+	defer conn.Close()
+	require.NoError(t, err)
+
+	client := api.NewCfgMgmtClient(conn)
+
+	cases := []struct {
+		description     string
+		runsProjects    []string
+		allowedProjects []string
+		runIsReturned   bool
+	}{
+		{
+			description:     "Run is returned when the one run project matches the one allowed project",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"one"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "No run is returned when the one run project does not match the one allowed project",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"two"},
+			runIsReturned:   false,
+		},
+		{
+			description:     "Run is returned when the one run project matches one of the two allowed projects",
+			runsProjects:    []string{"one"},
+			allowedProjects: []string{"one", "two"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is returned when both of the two run projects match both of the allowed projects",
+			runsProjects:    []string{"one", "two"},
+			allowedProjects: []string{"one", "two"},
+			runIsReturned:   true,
+		},
+		{
+			description:     "Run is retuned when the run does not have an assigned project and the allowed project is unassigned",
```

Comment:
```suggestion
			description:     "Run is returned when the run does not have an assigned project and the allowed project is unassigned",
```

Reply:
Thank you for taking a look. I updated the above problems.